src/main.rs

```diff
@@ -301,16 +316,21 @@ fn main() {
         let signal = loop {
             let signal = future::poll_fn(|| signals.poll());
             let crash = future::poll_fn(|| graceful_crash.poll());
+            let change = future::poll_fn(|| config_watcher.poll());
 
             let next = signal
                 .select2(crash)
+                .select2(change)
                 .wait()
                 .map_err(|_| ())
-                .expect("Neither stream errors");
+                .expect("None stream errors");
 
             let signal = match next {
-                future::Either::A((signal, _)) => signal.expect("Signal streams never end"),
-                future::Either::B((_crash, _)) => SIGINT, // Trigger graceful shutdown if a component crashed
+                future::Either::A((future::Either::A((signal, _)), _)) => {
```

Comment:
We should coordinate with #1219 to make sure that we don't regress here.

Reply:
Good catch, then I'll also add this for Windows code.